File: PetShop.Model/NarudzbaProizvod.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace PetShop.Model
{
    public class NarudzbaProizvod
    {
        public int NarudzbaId { get; set; }
        public int ProizvodId { get; set; }
        public int Kolicina { get; set; }

        public virtual Narudzba Narudzba { get; set; }
        public virtual Proizvod Proizvod { get; set; }
    }
}

File: petshop_mobile/lib/models/NarudzbaProizvod.dart

import 'package:petshop_mobile/models/Proizvod.dart';

class NarudzbaProizvod {
  final int? id;
  final int? narudzbaId;
  final int? proizvodId;
  final int kolicina;
  final Proizvod? proizvod;

  NarudzbaProizvod(
      {
        required this.id,
        required this.narudzbaId,
        required this.proizvodId,
        required this.kolicina,
        required this.proizvod,
      });

  factory NarudzbaProizvod.fromJson(Map<String, dynamic> json) {
    return NarudzbaProizvod(
      id: int.parse(json["id"].toString()),
      narudzbaId: int.parse(json["narudzbaId"].toString()),
      proizvodId: int.parse(json["proizvodId"].toString()),
      kolicina: int.parse(json["kolicina"].toString()),
      proizvod: Proizvod.fromJson(json["proizvod"]),
    );
  }

  Map<String, dynamic> toJson() => {
    "narudzbaId": narudzbaId,
    "proizvodId": proizvodId,
    "kolicina": kolicina,
    "proizvod": proizvod
  };
}
